### PR TITLE
Extract episode numbers from notation like S02E05

### PR DESCRIPTION
**Pull request by Self-Perfection**
Such notation seems to be widespread.

Also I'd like to note that this massive regex logic is hard to understand and debug. It really looks like this AnimeInfoExtractor should have unit tests.

### CODE REVIEW

File: trackma/extras/AnimeInfoExtractor.py [hunk 1]
```diff
@@ -258,9 +258,9 @@ def __extractEpisodeNumbers(self, filename):
         if not self.episodeStart:
             # Check if there is an episode specifier
             m = re.search(
-                '(?:[^0-9a-zA-Z])(E\.?|Ep(?:i|isode)?(?: |\.)?)(\d{1,}(?:\.\d{1})?)(?:[^\d]|$)', filename, flags=re.IGNORECASE)
+                '(?:[^0-9a-zA-Z])(S[0-9]+)?(E\.?|Ep(?:i|isode)?(?: |\.)?)(\d{1,}(?:\.\d{1})?)(?:[^\d]|$)', filename, flags=re.IGNORECASE)
```
Comment (ahmubashshir) [inline trackma/extras/AnimeInfoExtractor.py:261]: 👍🏽

**Comment by ahmubashshir** [comment]
> Such notation seems to be widespread.
> 
> Also I'd like to note that this massive regex logic is hard to understand and debug. It really looks like this AnimeInfoExtractor should have unit tests.

Not that massive though :3

**Comment by Self-Perfection** [comment]
Whoa, I've noticed there is big #547  with refactoring and test. It also includes  S01E01 parsing.
Well apparently this PR becomes obsolete if #547 get merged.

**Comment by z411** [comment]
#547 got merged. Thank you!

**Pull request closed without merging**